src/wallet/bdk.rs

```diff
@@ -308,77 +366,205 @@ impl Wallet {
                 let locktime = bond.lock_time;
                 println!(
                     "[{}] locktime={}",
-                    Address::from_script(&bond.script_pub_key(), self.store.network).unwrap(),
+                    Address::from_script(&bond.script_pub_key(), self.network).unwrap(),
                     locktime
                 );
             }
         }
         Ok(())
     }
 
-    pub fn init(
+    pub fn init<E: IntoWalletDescriptor + Hash + Debug>(
         path: &PathBuf,
-        rpc_config: &RPCConfig,
         seedphrase: String,
         passphrase: String,
-    ) -> Result<Self, WalletError> {
+        network: Network,
+        external_descriptor: E,
+        internal_descriptor: E,
+        fidelity_descriptor: E,
+        swapcoin_descriptor: E,
+        contract_descriptor: E,
+        rpc_config: &RPCConfig,
+        wallet_birthday: Option<u64>,
+    ) -> Result<Self, NewError> {
+        let bdk_network = convert_bitcoin_to_bdk_network(network);
+
         let file_name = path
             .file_name()
             .expect("file name expected")
             .to_str()
             .expect("expected")
             .to_string();
-        let rpc = Client::try_from(rpc_config)?;
-        let wallet_birthday = rpc.get_block_count()?;
         let store = WalletStore::init(
             file_name,
             path,
-            rpc_config.network,
+            network,
             seedphrase,
             passphrase,
-            Some(wallet_birthday),
-        )?;
-        Ok(Self {
-            rpc,
-            wallet_file_path: path.clone(),
+            wallet_birthday,
+        )
+        .expect("failed to initialise the walletstore");
+
+        let genesis_hash = genesis_block(bdk_network).block_hash();
+
+        let secp = bdk_Secp256k1::new();
+        let (chain, chain_changeset) = LocalChain::from_genesis_hash(genesis_hash);
+        let mut index = KeychainTxOutIndex::<KeychainKind>::default();
+
+        let signers = Wallet::create_signers(
+            &mut index,
+            &secp,
+            external_descriptor,
+            internal_descriptor,
+            fidelity_descriptor,
+            swapcoin_descriptor,
+            contract_descriptor,
+            network,
+        )
```

Comment:
Continuation of the previous comment..

We need to calculate the different descriptors here in the function from the master key. Then add them to the signer and other places.